src/API.jl

```diff
@@ -894,15 +894,33 @@ end
 
 precompile() = precompile(Context())
 function precompile(ctx::Context)
+    
+    is_stdlib_from_name(name::String) = name in values(stdlibs())
+    
     printpkgstyle(ctx, :Precompiling, "project...")
+    
+    num_tasks = parse(Int, get(ENV, "JULIA_NUM_PRECOMPILE_TASKS", string(Sys.CPU_THREADS + 1)))
```

Comment:
Seems kinda excessive to introduce an env variable for this. Its so specific.

Reply:
I'm not sure how else to gate this.  Suggestions?  There are some concerns that with a large core count this could accidentally OOM.

Reply:
How much memory does each worker use approximately. Isn't this the case for every parallel workload?

Reply:
> Isn't this the case for every parallel workload that uses memory?

Yes, but most other workloads allow tuning (e.g. via `-t`).

> I guess we should look at nthreads but everyone runs with that equal to 1.

There's that, and also Lyndon's comment above that this is more like a multiprocessing thing than a multithreading thing.  I also agree that I shouldn't have to limit my computation's thread count to limit the precompilation, and vice versa.

Reply:
I'd rather have it as a normal argument to the `precompile` function then. This is exactly what we already have to limit parallelism in the asynchronous package downloader.

Looking at it, funnily enough we do have an env variable for the package downloader but that seems like it was added as a workaround for something:

https://github.com/JuliaLang/Pkg.jl/blob/ede7b07f7bbd8b275b5d80660c19a79b82330ab6/src/Types.jl#L329-L331

Reply:
Although if we want to run this automatically when a package is updated, there is no chance to give this argument.

Perhaps there should be a `.julia/config/PkgConfig.toml` where things like this could be set?

Reply:
Anyway, let's go with this for now. Can always tweak it later.